docs/developer/designs/hierarchical-podgroup/README.md

```diff
@@ -0,0 +1,236 @@
+# Hierarchical PodGroup structure
+
+## Overview
+This document outlines an enhancement to the PodGroup architecture by introducing SubGroups within a PodGroup, enabling fine-grained control over gang scheduling and topology-aware placement for subsets of pods within a workload. While the PodGroup continues to enforce gang scheduling for the workload as a whole, this extension allows workloads to specify minMember and topology constraints on individual SubGroups, providing greater resolution over how smaller groups of pods are placed and scheduled. For example, a SubGroup can specify that only a subset of its pods must be scheduled together (minMember) and define specific network topology constraints aligned with that subset’s requirements. This design preserves atomic workload scheduling semantics while allowing advanced, granular placement policies that improve alignment with topology-aware scheduling strategies.
+
+## Motivation
+Modern inference workloads consist of components with varying network sensitivity and resource requirements, necessitating the ability to express fine-grained scheduling and placement preferences within the context of gang-scheduled workloads. Enforcing uniform constraints at the PodGroup level can lead to overly restrictive placement requirements that do not reflect the nuanced needs of different workload components. By introducing SubGroups within a PodGroup, workloads can specify per-SubGroup minMember requirements and topology constraints while maintaining overall gang scheduling semantics. This enables advanced placement strategies, improves scheduling flexibility and efficiency, and ensures that differentiated placement and partial gang constraints can be expressed without relaxing the workload’s atomic scheduling guarantees.
+
+## Goals
```

Comment:
IMO, an important goal is to not have a regression in the *experience* for existing, simpler use cases. Backwards compatibility is mentioned below, so no features are expected to be lost, but it's also important to keep the experience good as well - I mainly think of errors, conditions and logs, not becoming too complicated for simple 1-subgroup podgroups